packages/material-ui/src/Tabs/Tabs.js

```diff
@@ -488,14 +500,27 @@ const Tabs = React.forwardRef(function Tabs(inProps, ref) {
       return;
     }
 
-    if (tabMeta[start] < tabsMeta[start]) {
-      // left side of button is out of view
-      const nextScrollStart = tabsMeta[scrollStart] + (tabMeta[start] - tabsMeta[start]);
-      scroll(nextScrollStart, { animation });
-    } else if (tabMeta[end] > tabsMeta[end]) {
-      // right side of button is out of view
-      const nextScrollStart = tabsMeta[scrollStart] + (tabMeta[end] - tabsMeta[end]);
-      scroll(nextScrollStart, { animation });
+    if (isRtl) {
+      if (tabMeta[end] < tabsMeta[end]) {
+        // left side of button is out of view
+        const nextScrollStart = tabsMeta[scrollStart] + (tabMeta[end] - tabsMeta[end]);
+        scroll(nextScrollStart, { animation });
+      } else if (tabMeta[start] > tabsMeta[start]) {
+        // right side of button is out of view
+        const nextScrollStart = tabsMeta[scrollStart] + (tabMeta[start] - tabsMeta[start]);
+        scroll(nextScrollStart, { animation });
+      }
+    } else {
+      // eslint-disable-next-line no-lonely-if
+      if (tabMeta[start] < tabsMeta[start]) {
+        // left side of button is out of view
+        const nextScrollStart = tabsMeta[scrollStart] + (tabMeta[start] - tabsMeta[start]);
+        scroll(nextScrollStart, { animation });
+      } else if (tabMeta[end] > tabsMeta[end]) {
+        // right side of button is out of view
+        const nextScrollStart = tabsMeta[scrollStart] + (tabMeta[end] - tabsMeta[end]);
+        scroll(nextScrollStart, { animation });
+      }
```

Comment:
I might have a solution that doesn't require to touch this part:

```diff
```diff
diff --git a/packages/material-ui/src/Tabs/Tabs.js b/packages/material-ui/src/Tabs/Tabs.js
index 68273478c1..5189a3e290 100644
--- a/packages/material-ui/src/Tabs/Tabs.js
+++ b/packages/material-ui/src/Tabs/Tabs.js
@@ -356,30 +356,34 @@ const Tabs = React.forwardRef(function Tabs(inProps, ref) {
   const updateIndicatorState = useEventCallback(() => {
     const { tabsMeta, tabMeta } = getTabsMeta();
     let startValue = 0;
+    let startIndicator;

     if (tabMeta && tabsMeta) {
       if (vertical) {
+        startIndicator = 'top';
         startValue = tabMeta.top - tabsMeta.top + tabsMeta.scrollTop;
       } else {
+        startIndicator = isRtl ? 'right' : 'left';
         const correction = isRtl
           ? tabsMeta.scrollLeftNormalized + tabsMeta.clientWidth - tabsMeta.scrollWidth
           : tabsMeta.scrollLeft;
-        startValue = tabMeta.left - tabsMeta.left + correction;
+        startValue =
+          (isRtl ? -1 : 1) * (tabMeta[startIndicator] - tabsMeta[startIndicator] + correction);
       }
     }

     const newIndicatorStyle = {
-      [start]: startValue,
+      [startIndicator]: startValue,
       // May be wrong until the font is loaded.
       [size]: tabMeta ? tabMeta[size] : 0,
     };

     // IE11 support, replace with Number.isNaN
     // eslint-disable-next-line no-restricted-globals
-    if (isNaN(indicatorStyle[start]) || isNaN(indicatorStyle[size])) {
+    if (isNaN(indicatorStyle[startIndicator]) || isNaN(indicatorStyle[size])) {
       setIndicatorStyle(newIndicatorStyle);
     } else {
-      const dStart = Math.abs(indicatorStyle[start] - newIndicatorStyle[start]);
+      const dStart = Math.abs(indicatorStyle[startIndicator] - newIndicatorStyle[startIndicator]);
       const dSize = Math.abs(indicatorStyle[size] - newIndicatorStyle[size]);

       if (dStart >= 1 || dSize >= 1) {
```
```

Reply:
Yes, this is better 👍